src/KubernetesClient/Kubernetes.Generic.cs

```diff
@@ -0,0 +1,1131 @@
+            }
+            if (orphanDependents != null)
+            {
+                queryParameters.Add($"orphanDependents={System.Uri.EscapeDataString(SafeJsonConvert.SerializeObject(orphanDependents, SerializationSettings).Trim('"'))}");
+            }
+            if (propagationPolicy != null)
+            {
+                queryParameters.Add($"propagationPolicy={System.Uri.EscapeDataString(propagationPolicy)}");
+            }
+            if (pretty != null)
+            {
+                queryParameters.Add($"pretty={System.Uri.EscapeDataString(pretty)}");
+            }
+            if (queryParameters.Count > 0)
+            {
+                url += "?" + string.Join("&", queryParameters);
+            }
+            // Create HTTP transport objects
+            var httpRequest = new HttpRequestMessage();
+            HttpResponseMessage httpResponse = null;
+            httpRequest.Method = new HttpMethod("DELETE");
+            httpRequest.RequestUri = new System.Uri(url);
+            // Set Headers
+
+
+            if (customHeaders != null)
+            {
+                foreach(var header in customHeaders)
+                {
+                    if (httpRequest.Headers.Contains(header.Key))
+                    {
+                        httpRequest.Headers.Remove(header.Key);
+                    }
+                    httpRequest.Headers.TryAddWithoutValidation(header.Key, header.Value);
+                }
+            }
+
+            // Serialize Request
+            string requestContent = null;
+            if(body != null)
+            {
+                requestContent = SafeJsonConvert.SerializeObject(body, SerializationSettings);
+                httpRequest.Content = new StringContent(requestContent, System.Text.Encoding.UTF8);
+                httpRequest.Content.Headers.ContentType =System.Net.Http.Headers.MediaTypeHeaderValue.Parse("application/json; charset=utf-8");
+            }
+            // Set Credentials
+            if (Credentials != null)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                await Credentials.ProcessHttpRequestAsync(httpRequest, cancellationToken).ConfigureAwait(false);
+            }
+            // Send Request
+            if (shouldTrace)
+            {
+                ServiceClientTracing.SendRequest(invocationId, httpRequest);
+            }
+            cancellationToken.ThrowIfCancellationRequested();
+            httpResponse = await HttpClient.SendAsync(httpRequest, cancellationToken).ConfigureAwait(false);
+            if (shouldTrace)
+            {
+                ServiceClientTracing.ReceiveResponse(invocationId, httpResponse);
+            }
+            var statusCode = httpResponse.StatusCode;
+            cancellationToken.ThrowIfCancellationRequested();
+            string responseContent = null;
+            if (statusCode != OK && statusCode != Accepted && statusCode != Unauthorized)
```

Comment:
Delete can actually return Accepted and a `Status` object instead of the deleted object. We should probably handle that case here too.

Reply:
This code was taken directly from generated methods of Kubernetes object (hence the duplicate code - I wasn't trying to pretty it up beyond making it generic). 

Personally I'm of the opinion that once these methods are standardized, 90% of Kubernetes class autogenerated methods can be made to just call these generic methods.